Remove label component

bethcollins92 opened the pull request:
## Done

- Remove all docs and examples of the deprecated `p-label` pattern
- Replace all instances on the site with chip pattern
- Add to what's new table
- Add to the upgrade guide
- Added text colour to chips 

Fixes

#4140 


## QA

- Open [demo](https://vanilla-framework-4154.demos.haus)
- Check the label component no longer exists in the docs/examples
- Review updated documentation:
  - Check what's new table as these labels have all been updated to use the chip
  - Check the 3.0 upgrade guide as a label section was added

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

webteam-app commented:
Demo starting at https://vanilla-framework-4154.demos.haus

bethcollins92 commented:
@bartaz I added the text colour for the chips

anthonydillon commented on scss/_patterns_chip.scss:
```diff
@@ -19,6 +19,7 @@
 
     .p-chip__lead,
     .p-chip__value {
+      color: $colors--light-theme--text-default;
```
Would it be better to set this `color` in the `vf-chip-type` mixin when the background-color is set?

bethcollins92 replied:
I thought as the text colour is always going to be the same and it isn't dynamic it made sense to set it with the rest of the text styles? But happy to change it just want to understand why

anthonydillon replied:
Probably fine but my thought is that we might have variants that require adaptive colour based on the background.

bartaz replied:
Currently all have same color, but when we introduce light/dark themes we will move the color setting to themes.

I think it's fine for now.

bartaz commented on package.json:
```diff
@@ -29,6 +29,7 @@
     "start": "yarn build && yarn serve",
     "build": "yarn build-scss && yarn build-js",
     "build-scss": "yarn run build-js && sass --load-path=node_modules --embed-sources --style=compressed scss:build/css && postcss --use autoprefixer --replace 'build/css/**/*.css' --map",
+    "build:essential": "yarn run build-js && sass --load-path=node_modules --embed-sources --style=compressed scss/build.scss:build/css/build.css scss/docs:build/css/docs && postcss --use autoprefixer --replace 'build/css/**/*.css' --map",
```
Not sure why it's here. It was merged separately.
Could you rebase on top of fresh `main` branch?

bartaz commented on templates/docs/base/code.md:
```diff
@@ -102,7 +102,9 @@ View example of the code snippet with syntax highlighting
 ## Bordered
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
Remove it please. Code is no longer "new" pattern.

bartaz commented on templates/docs/base/forms.md:
```diff
@@ -204,7 +204,9 @@ View example of the dense form elements
 ## Password toggle
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
Remove this one as well. It's not new in 3.0.

bartaz commented on templates/docs/layouts/fluid-breakout.md:
```diff
@@ -23,7 +23,9 @@ The wrapper aims to align as much as possible with the 12 column grid. On smalle
 ### Toolbar
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
Remove that one as well.

bartaz commented on templates/docs/patterns/buttons.md:
```diff
@@ -58,7 +58,9 @@ View example of the brand button pattern
 ## Link
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
And this one can be dropped.

bartaz commented on templates/docs/patterns/links.md:
```diff
@@ -44,7 +44,9 @@ View example of the back to top pattern
 ## Skip link
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
Remove it plz.

bartaz commented on templates/docs/patterns/notification.md:
```diff
@@ -14,7 +14,9 @@ Notifications are used to attract the user's attention. They offer four separate
 ### Information
 
-<span class="p-label--updated">Updated</span>
+<div class="p-chip--information">
```
Remove those as well, all of them.

bartaz commented on templates/docs/patterns/search-and-filter.md:
```diff
@@ -6,7 +6,9 @@ context:
 # Search and filter
 
-<span class="p-label--new">New</span>
+<div class="p-chip--positive">
```
Oh, it's been a while since that was "new".

bartaz commented:
@bethcollins92 Since we removed labels page, I think we should redirect it to chips page instead. Could you add a redirect to `redirects.yaml` file?

bethcollins92 commented:
@bartaz Thanks, made suggested changes

lyubomir-popov commented:
![image](https://user-images.githubusercontent.com/2741678/143604837-a759d785-a952-4222-98de-9bfc86a9a61b.png)
Minimal adjustments (borders only, so they meet contrast ratio requirements against `#333` and similar backgrounds like the jaas one)

bartaz commented on templates/docs/patterns/slider.md:
```diff
@@ -15,7 +15,9 @@ when you want to provide a numeric representation of the slider value, as well a
 </a></div>
 
-<span class="p-label--updated">Updated</span> In version 2.6.0 Vanilla framework added slider styling as default for
+<div class="p-chip--information is-inline is-dense">
```
I know our examples show them as divs (which is wrong I think), but we should use spans here. Otherwise divs break out of paragraphs (because you can't put `<div>` in `<p>`):

<img width="708" alt="Screenshot 2021-11-26 at 17 06 58" src="https://user-images.githubusercontent.com/83575/143607436-4cfb4475-1fb3-43fb-a8c9-805718db4906.png">


```suggestion
<span class="p-chip--information is-inline is-dense">
```

bartaz commented on templates/docs/patterns/slider.md:
```diff
@@ -15,7 +15,9 @@ when you want to provide a numeric representation of the slider value, as well a
-<span class="p-label--updated">Updated</span> In version 2.6.0 Vanilla framework added slider styling as default for
+<div class="p-chip--information is-inline is-dense">
+  <span class="p-chip__value">Updated</span>
+</div>In version 2.6.0 Vanilla framework added slider styling as default for
```
```suggestion
</span> In version 2.6.0 Vanilla framework added slider styling as default for
```

bartaz commented on templates/docs/upgrade-guide-v3.md:
```diff
@@ -142,6 +142,19 @@ List of removed max width features includes:
+| `.p-label--new`         | `.p-chip--positive`    |
+| `.p-label--updated`     | `.p-chip--information` |
+| `.p-label--validated`   | `.p-chip--information` |
+
```
We should add a sentence about `vf-p-label` mixin being removed and if anyone uses it, it should be removed from code.

We actually removed couple mixins (all from labels file) `vf-p-label-new` `-updated` `-deprecated` `-in-progress` and `-validated`

bartaz reviewed (commented):
Couple other small fixes needed

bartaz commented on redirects.yaml:
```diff
@@ -41,6 +41,7 @@
 /docs/patterns/link: /docs/patterns/links
 /docs/patterns/top: /docs/patterns/links#back-to-top
 /docs/patterns/list: /docs/patterns/lists
+/docs/patterns/labels: /docs/patterns/chips
```
It's `/chip`, we are a bit inconsistent in naming pages unfortunately

```suggestion
/docs/patterns/labels: /docs/patterns/chip
```

bartaz commented on templates/docs/upgrade-guide-v3.md:
```diff
@@ -142,6 +142,21 @@ List of removed max width features includes:
 
+## Labels
+
+The label pattern has been removed, and we've added tinted chips. The tints are based on the semantic colours (positive, caution, negative) plus a dark blue one that matches the blue used in the information flavour of the notification component. The default chip colour (grey) is now referred to as "neutral". Please read the [chip documentation](/docs/patterns/chips) to see how the markup and clas ...
```
This still fails linkchecker.

```suggestion
The label pattern has been removed, and we've added tinted chips. The tints are based on the semantic colours (positive, caution, negative) plus a dark blue one that matches the blue used in the information flavour of the notification component. The default chip colour (grey) is now referred to as "neutral". Please read the [chip documentation](/docs/patterns/chip) to see how the markup and class names have changed, including the addition of `p-chip__lead` and `p-chip__value` classes. In places where you're replacing the `p-label` you'll probably need to add the `is-dense` modifier.
```

bartaz reviewed (approved):
LGTM, but you need to fix one more link.

bartaz commented:
Not sure if it needs design review, we have now separate issue for dark theme: #4158

bartaz commented:
Just noticed in percy:

<img width="793" alt="Screenshot 2021-11-29 at 08 10 45" src="https://user-images.githubusercontent.com/83575/143823625-ff36808c-f74a-4c95-83eb-906561ebe007.png">

By forcing text color on chip we also overridden the color of 'lead' part of chip ("Owner:" …) - it is meant to be using muted heading color. Maybe it's a css specificity issue?

bethcollins92 commented:
Thanks @bartaz, I've moved the color selector so it's only targetting the `p-chip__value` and not the `p-chip__lead` and fixed the link. Spoken to Lyubo and we can address the dark theme in the other issue

bartaz commented:
The chips in side navigation are not aligned with the icons on the right. It seems that chips have some small right margin.
I guess `is-inline` should have margin removed?

<img width="492" alt="Screenshot 2021-12-01 at 10 58 39" src="https://user-images.githubusercontent.com/83575/144213698-43c4437d-3d79-4037-8d0b-b20b382d452a.png">

bethcollins92 commented:
@bartaz Removed margin from `is-inline`

bartaz reviewed (approved):
LGTM

lyubomir-popov commented:
Sorry one last change - 	can we replace these with dense ones - if in the side nav theyu should always be as small as possible:
![image](https://user-images.githubusercontent.com/2741678/144276981-42589d9f-1036-46dc-8605-3bd198770146.png)

lyubomir-popov commented:
just noticed this, can you please make the chips in the application layout dark please?

bethcollins92 commented:
@lyubomir-popov I changed them to the dense version could you review/approve percy please?

bethcollins92 commented:
@lyubomir-popov I've rebased and updated the chips on dark backgrounds
